### ISSUE TITLE
Research: create a template with photonOS

### ISSUE THREAD

**Issue by alexellis** [openfaas/faas#886]
As a research item I'd like to see someone port two of our templates from Alpine Linux to PhotonOS from VMware. This would not replace our community templates, but would provide an experimental version of 1-2 programming languages in our incubator repository.

I could see this as a positive step to helping enterprises bring OpenFaaS to serverless to whatever "golden Linux" they have been given enterprise support for - whether from RedHat, VMware or another vendor like SuSE.

PhotonOS seems like a good place to start given access to colleagues at VMware and the potential to use try it out internally and get feedback. 

https://docs.docker.com/samples/library/photon/

## Task 

Rebuild one of the most popular templates for PhotonOS i.e. Python or Go using the base image.

Alex

**Comment by padiazg** [comment]
I want to take this. That's the kind of stuff I like to do.

**Comment by padiazg** [comment]
I managed to port 3 of the templates to PhotonOS, pyton2, python3 and node.js. They are [here](https://github.com/padiazg/of-photon-templates) if anyone wants to give them a try.

Found that it's hard to port a template using the Alpine "recipe", that is installing the language from source and compiling it.

After comparing the latest source code version with the version provided as packages for each language, found that they match pretty close, so I decided to write the templates to use packages installed from the official PhotonOS repository.

The major drawback I found is that PhotonOS  delegates the security management of a container to an external server., which is very small an quick to install, but it can't run on a container so it must be installed in a VM. 

To create a user or group you use pmd_cli (https://github.com/vmware/photon/blob/master/docs/pmd-cli.md) and refer to the management daemon.

Due to this fact, I wrote the templates to run as the root user, and the non-root user approach is not available here. This seems very impractical, I think a step backwards.

I am also worried that I could not compile a single language from its source code. That indicates that special care must be taken for this.
